openedx/adg/lms/applications/analytics.py

```diff
@@ -0,0 +1,191 @@
+    Waitlisted and Accepted
+
+    Arguments:
+        user (User): User to emit progress update for
+    """
+    application_progress = user.application_hub.application_progress
+
+    application_status = APPLICATION_PROGRESS.get(application_progress, APPLICATION_PROGRESS[APPLICATION_STARTED])
+    segment.identify(user.id, {'applicationStatus': application_status})
+
+    application_event = APPLICATION_EVENTS.get(application_progress, APPLICATION_EVENTS[APPLICATION_STARTED])
+    event_properties = {
+        'applicationStatus': application_status
+    }
+    if hasattr(user, 'application'):
+        event_properties['applicationId'] = user.application.id
+    segment.track(user.id, application_event, event_properties)
+
+
+def emit_user_info_update(user, application_step):
+    """
+    Emit an event to update the user information at Segment after different steps of the application.
+
+    Arguments:
+        user (User): User for which to emit the event
+        application_step (str): Step of the application to update the user information for
+    """
+    if application_step == CONTACT_INFO:
+        attributes = {
+            'isSaudi': user.extended_profile.saudi_national,
+            'age': user.extended_profile.age,
+            'heardAboutUs': user.extended_profile.hear_about_omni,
+        }
+
+    elif application_step == EXPERIENCE:
+        first_education = user.application.applications_educations.first()
+        education_start_date, education_end_date = _get_start_date_and_end_date(first_education)
+        attributes = {
+            'school': first_education.name_of_school,
+            'degree': first_education.get_degree_display(),
+            'educationStartDate': education_start_date,
+            'educationEndDate': education_end_date,
+        }
+
+        first_experience = user.application.applications_workexperiences.first()
+        if first_experience:
+            experience_start_date, experience_end_date = _get_start_date_and_end_date(first_experience)
+            attributes.update({
+                'workOrganization': first_experience.name_of_organization,
+                'jobTitle': first_experience.job_position_title,
+                'workStartDate': experience_start_date,
+                'workEndDate': experience_end_date
+            })
+
+    else:
+        attributes = {
+            'businessLine': user.application.business_line.title
+        }
+        emit_application_progress(user)
+
+    segment.identify(user.id, attributes)
+    segment.track(user.id, APPLICATION_EVENTS.get(application_step), attributes)
+
+
+def _get_start_date_and_end_date(position):
+    """
+    Check if the position is ongoing (current), and returns the start_date and the end_date of the
+    position.
+
+    Arguments:
+        position (UserStartAndEndDates): WorkExperience or Education to get the start and end date for
```

Comment:
I guess the data type for position is wrong

Reply:
Both `WorkExperience` and `Education` inherit from `UserStartAndEndDates` class. And we are using the dates that come from the `UserStartAndEndDates`. Therefore, this function can work for any `UserStartAndEndDates` object